testing/playwright/e2e/downstream/plan-power-state.spec.ts

```diff
@@ -0,0 +1,133 @@
+    await expect(planDetailsPage.detailsTab.savePowerStateButton).toBeDisabled();
+    await planDetailsPage.detailsTab.targetPowerStateSelect.click();
+    await expect(planDetailsPage.detailsTab.powerStateOptionAuto).toHaveText(
+      'Retain source VM power state',
+    );
+    await expect(planDetailsPage.detailsTab.powerStateOptionOn).toHaveText('Powered on');
+    await expect(planDetailsPage.detailsTab.powerStateOptionOff).toHaveText('Powered off');
+
+    // Select new power state and save
+    const detailsNewPowerState = 'off';
+    await planDetailsPage.detailsTab.powerStateOption(detailsNewPowerState).click();
+    await expect(planDetailsPage.detailsTab.savePowerStateButton).toBeEnabled();
+    await planDetailsPage.detailsTab.savePowerStateButton.click();
+
+    // Verify the change is reflected on the details page
+    await expect(planDetailsPage.detailsTab.targetVMPowerState('Powered off')).toBeVisible();
+
+    // Step 5: Navigate to VMs tab and verify column management
+    await planDetailsPage.virtualMachinesTab.navigateToVirtualMachinesTab();
+
+    // Enable target power state column
+    await planDetailsPage.virtualMachinesTab.enableColumn('Target power state');
+
+    // Verify column is visible
+    const isPowerStateColumnVisible =
+      await planDetailsPage.virtualMachinesTab.isColumnVisible('Target power state');
+    expect(isPowerStateColumnVisible).toBe(true);
+
+    // Step 6: Verify VM inherits plan power state
+    const vmName = testData.virtualMachines?.[0]?.sourceName ?? '';
+    const powerState = await planDetailsPage.virtualMachinesTab.getVMPowerState(vmName);
+    expect(powerState).toBe('Powered off (Inherited from plan)');
+
+    // Step 7: Edit VM power state and verify all options
+    await planDetailsPage.virtualMachinesTab.openPowerStateDialog(vmName);
+
+    // Verify save is disabled and all 4 options are present
+    const { virtualMachinesTab } = planDetailsPage;
+    await expect(virtualMachinesTab.powerStateModalSaveButton).toBeDisabled();
+
+    // Click dropdown to see all options
+    await virtualMachinesTab.powerStateModalSelect.click();
+    await expect(virtualMachinesTab.powerStateOptionInherit).toBeVisible();
+    await expect(virtualMachinesTab.powerStateOptionInherit).toContainText('Set to: Powered off');
+    await expect(virtualMachinesTab.powerStateOptionAuto).toBeVisible();
+    await expect(virtualMachinesTab.powerStateOptionOn).toBeVisible();
+    await expect(virtualMachinesTab.powerStateOptionOff).toBeVisible();
+
+    // Step 8: Change VM power state and verify
+    await virtualMachinesTab.powerStateOptionOn.click();
+    await expect(virtualMachinesTab.powerStateModalSaveButton).toBeEnabled();
+    await virtualMachinesTab.powerStateModalSaveButton.click();
+
+    // Wait for modal to close and table to refresh
+    await expect(virtualMachinesTab.editTargetPowerStateModal).not.toBeVisible();
+    await page.waitForTimeout(1000); // Give time for the table to update
```

Comment:
We should avoid arbitrary timeouts. Is there not an element or request we can wait for after the modal closes?